addon-chrome/ChromePlacesProvider.js

```diff
@@ -0,0 +1,379 @@
+          // api uses start and end time as OR instead of AND
+          // so filter it for now or we could just use this ???
+          results = results.filter((result) => result.lastVisitTime > startTime && result.lastVisitTime < endTime);
+        }
+        this._getBookmark()
+          .then((bookmarks) => {
+            const histories = results.map((result) => this._transformHistory(result, bookmarks));
+            this._filterBlockedUrls(histories).then(resolve);
+          });
+      });
+    });
+  }
+
+  /**
+   * Rank history items (excluding google searches and localhost sites) by frecency score, then lastVisitTime, then url
+   *
+   * @param {Array} histories - List of history items
+   *
+   * @returns {Object} Promise that resolves with list of history items ranked by their frencency, lastVisitTime, url
+   */
+  static _getTopFrecentSites(histories) {
+    return new Promise((resolve) => {
+      const rows = histories
+        .filter((hist) => !/(google|localhost)/.test(hist.url))
+        .map((hist) => {
+          const frencency = this._calculateFrecency(hist);
+          return Object.assign(hist, {frencency});
+        })
+        .sort((a, b) => {
+          if (a.frencency > b.frencency) {
+            return -1;
+          }
+          if (a.frencency < b.frencency) {
+            return 1;
+          }
+          if (a.lastVisitTime > b.lastVisitTime) {
+            return -1;
+          }
+          if (a.url > b.url) {
+            return -1;
+          }
+          return 0;
+        });
+
+      resolve(rows);
+    });
+  }
+
+  /**
+   * Calculate frecency scores a history items base on
+   * https://dxr.mozilla.org/mozilla-central/source/mobile/android/base/java/org/mozilla/gecko/db/BrowserContract.java#124
+   * numVisits * max(1, 100 * 225 / (age*age + 225))
+   *
+   * @param {Object} hist History item to calculate frecency score for
+   *
+   * @returns {Number} The calculated frecency for the history item
+   */
+  static _calculateFrecency(hist) {
+    const microsecondsPerDay = 86400000000;
+    const age = (new Date().getTime() - hist.lastVisitDate) / microsecondsPerDay;
+    const frencency = hist.visitCount * Math.max(1, 100 * 225 / (age * age + 225));
+    return frencency;
+  }
+
+  /**
+   * Get a list of items from history and bookmarks that is a highlight item
+   * Shuffle everytime for higher chance of getting different sets of highlights
+   *
+   * @returns {Object} Promise that resolves with a list of highlight items
+   */
+  static _getHighlights() {
+    return new Promise((resolve) => {
+      const bookmarkPromise = this._getBookmark();
+      const historyPromise = this.getHistory();
+
+      Promise.all([bookmarkPromise, historyPromise]).then((results) => {
+        const bookmarks = results[0].filter((bookmarks) => !/(google|localhost)/.test(bookmarks.url));
+        const histories = results[1].filter((hist) => !/(google|localhost)/.test(hist.url));
```

Comment:
you're doing this type of filtering in a lot of places, skipping out on google|localhost urls. at the very least, take this out into a constant perhaps, to make it easy to edit across the app.